src/lua/print.lua

```diff
@@ -0,0 +1,15 @@
+local M = {}
+
+M.raw_print = print
+
+_G.print = function(...)
+    if M.before_cb ~= nil then
```

Comment:
Side note: `M.before_cb and M.before_cb()` looks more compact. Furthermore, callback call might through an error, so it might be worth to use `pcall` here. All in all, it works, so feel free to ignore these nits.

Reply:
It is an expression and I can't use it as a statement (`local _ = M.before_cb and M.before_cb()` would work). I'll keep it as is.

Those callbacks are simple now and I would prefer to leave the calls without `pcall()`. However I'll add a comment that it is expected that they will not raise errors.

```diff
diff --git a/src/lua/print.lua b/src/lua/print.lua
index 3dabe4b13..75fe436b7 100644
--- a/src/lua/print.lua
+++ b/src/lua/print.lua
@@ -2,6 +2,8 @@ local M = {}
 
 M.raw_print = print
 
+-- before_cb() and after_cb() must not raise an error, must not
+-- yield. It would break expectations about the print() function.
 _G.print = function(...)
     if M.before_cb ~= nil then
         M.before_cb()
```